### DISCUSSION TITLE
Multi site?

### DISCUSSION THREAD

**Discussion by Jarrodsz** [payloadcms/payload#1150] [Q&A]
1 payload install multiple sites? Is this possible?

Best react cms i hoped for long time dtarting back 2018

**Comment by jmikrut** [comment]
Hell yeah. We love hearing that.

Having multiple sites within one Payload instance is totally possible thanks to our super powerful access control. You'd basically build each piece of "content" to have a "Website" relationship field, and then build out your access control to only show the documents belonging to a certain website to an admin editor if they have permission to view the documents.

Would love to see what you build if you end up going down this path!

**Reply by Jarrodsz** [reply]
Thank you James. I will be starting with a simple website and see how it works out. Great this is a possibility. 
I have several marketing / startup websites that I could migrate to this solution in the end. Something on my Wishlist for a long time.

**Reply by Jarrodsz** [reply]
Perhaps you elaborate a little bit on how to set this up? Would this be a great marketing blog post perhaps? Most systems do not offert this can of flexibility. Else I will figure it out myself in the end, lets start with 1 website first ;)

**Reply by mtrach2020** [reply]
@jmikrut I was thinking of approaching multi-tenancy the same way until I realized a potential issue with that, though I might be missing something. Imagine you have multiple e-commerce stores and one individual is a customer of more than one of them and you want to have sign-in for customers. Wouldn't this create a conflict? It looks like once you designate something as an auth collection, Email can't be duplicated. Is there a way to let Payload know that it's fine for email in an auth collection to be duplicated if they have different website ID?

**Reply by jmikrut** [reply]
Hey @mtrach2020 — so, the way I'd do this is to have an `access` or `sites` relationship field that is tied to a Customer. And they can have more than one `site` granted to them in their profile. Then, you'd just write your access control accordingly!

This is totally possible but I would not duplicate customers. Instead, keep the same customer, and add further access to the one individual customer!

**Reply by mtrach2020** [reply]
Thanks for the quick response! There are a few data integrity issues I see with this but I'm also thinking of some potential solutions. By the way I apologize if I'm misunderstanding something here - I'm new to MongoDB so sometimes I'm still stuck in a "relational" mindset.

**Issue 1**
This would mean different tenants are sharing the same customer document. So what if a customer gave his address to one tenant but not the other - it wouldn't be ok for both of them to see that.

Maybe this is not that hard to solve, though: have a customerLogin and customerData collection, so customerLogin has no information other than email/password. Then customerData has a relationship with customerLogin but is scoped to only one tenant.

**Issue 2**
It would also be weird if someone went to a site they've never been before and were already logged in, or if they logged out of one site, then they'd be logged out of another.

But maybe that nots an issue either with same-site cookies?

**Issue 3**
You'd essentially have to turn off the ability for admins to make changes to customerLogin collection (e.g., email, unlocking, etc.) because it would effect the other tenants.

I can't really think of a way around this issue.

**Comment by DavidOliver** [comment]
I'm trying out Payload (liking it so far!) and was wondering about this for my clients' sites, but the setup and config seems pretty simple so I'm thinking having an install per project seems simpler overall, assuming that applying updates isn't a nightmare.

Having Caddy serve as a reverse proxy for multiple projects is simple enough, and I suppose common sets of config objects could be put into a JS dependency that's imported into each project as required?

**Reply by Jarrodsz** [reply]
still out of box solution will be more convincing

**Reply by jmikrut** [reply]
Hey @DavidOliver @Jarrodsz I actually covered a bit on multisite in the Access Control vid I just published a few weeks ago:

https://www.youtube.com/watch?v=DoPLyXG26Dg

Maybe this has some help for you?

I tend to agree with @DavidOliver though. Generally we set up separate sites because why not? It keeps things simpler. Especially if you have a server set up already with `nginx`.

**Reply by DavidOliver** [reply]
@jmikrut, thanks; that was a nice watch. Payload certainly seems flexible.

**Comment by Jarrodsz** [comment]
you could install it on a per site base, but still for me to much hassle. I want 1 central CMS to have all my projects in. 
An example on how to implement this, support in the core would be very welcome and would fill a gap since not many cms can do multi site.

I hope the core team will consider this implementing this native and documenting it. 
for me personally I won't spend time on payload until this feature is implemented.

**Comment by Jarrodsz** [comment]
What I would like to accomplish doing this is as follow:

- Create Landing pages for projects easily and manage them all from 1 central place.
- I would like to create 1 layout ( design ) and be able to easily change the content of it for several websites like chasing logo and change some parts of the site perhaps
- Have 1 central admin to manage all these sites

**Comment by Jarrodsz** [comment]
1 year later any updates on this?

i can use monorepro pnpm already use that but how about the orher issues🙏
